lib/json.c

```diff
@@ -485,7 +485,7 @@ sr_json_parse_ex(struct sr_json_settings *settings,
                         continue;
 
                     case '"':
-                        if (flags & flag_need_comma && (!state.settings.settings & SR_JSON_RELAXED_COMMAS))
+                        if (flags & flag_need_comma && !(state.settings.settings & SR_JSON_RELAXED_COMMAS))
```

Comment:
The commit message doesn’t mention anything about commas.

Reply:
This was just an unrelated detail pointed out by Clang. The `!` does not do what seems to have been intended and either a `~` should have been used instead or this parenthesization used.

I have removed this change, since it doesn't really matter if we only have one possible flag.

Reply:
I mean, if it’s correct, it’s fine to make that change, just in a separate commit.